### PR TITLE
Override package source

### PR DESCRIPTION
**Pull request by jaygeeseman**
Allows overriding the MongoDB package source location via attribute.

This is for https://www.pivotaltracker.com/story/show/97755750

### CODE REVIEW

File: recipes/default.rb [hunk 1]
```diff
@@ -27,15 +27,27 @@
   action :nothing
 end
 
+# Include :package_source attribute to download from an alternate location
+remote_file "#{Chef::Config[:file_cache_path]}/mongodb-10gen.deb" do
+  source node[:mongodb][:package_source]
```
Comment (mkantor) [inline recipes/default.rb]: Could you mention this attribute [in the README](https://github.com/Kabam/chef-mongodb#attributes)? Also maybe add it to attributes/default.rb (with a default value of `nil` or something).

**Comment by mkantor** [comment]
Nice workaround! This is much simpler than hosting our own repository (which is what I was thinking).

My only design suggestion would be to add a `package_checksum` attribute so that we can get `remote_file` to check if the package it downloads is legit.

File: recipes/default.rb (hunk 1 above)
Comment (mkantor) [inline recipes/default.rb]: Also possibly name this `:install_url` to match up with [the similar](https://github.com/Kabam/chef-mongodb/blob/8ededecb9120326effdd6e36d762e757375edd5d/attributes/mms_agent.rb#L4) [MMS attributes](https://github.com/Kabam/chef-mongodb/blob/8ededecb9120326effdd6e36d762e757375edd5d/attributes/mms_backup.rb#L4).

**Comment by FangAtKabam** [comment]
looks good

File: recipes/default.rb (hunk 1 above)
Comment (jaygeeseman) [inline recipes/default.rb]: I took all your suggestions here

**Comment by mkantor** [comment]
Any thoughts on the checksum idea?

**Comment by jaygeeseman** [comment]
A newer version of Chef has a `verify` attribute, but before that it's actually quite difficult to implement checksum verification considering the order of events we have can change.

TIL file checksum in Chef seems to be a political / touchy subject. For some reason.

**Comment by mkantor** [comment]
Huh, based on a closer reading of [the docs](https://docs.chef.io/resource_remote_file.html) I guess `checksum` is only used for caching purposes, and not verification. I assumed the resource would throw an error if its `source` pointed to a file which did not match the checksum.

Lame.

**Comment by mkantor** [comment]
LGTM.

**Pull request merged**